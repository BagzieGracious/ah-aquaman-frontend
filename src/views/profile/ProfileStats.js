import React from "react";
import { Link } from "react-router-dom";
import Moment from "react-moment";
import ReactTimeAgo from "react-time-ago";
import FollowingList from "./followingList";
import FollowersList from "./followersList";

const ProfileStats = (props) => {
  const { following, followers } = props;
  const list = props.following.following;
  const followersList = followers.followers;
  return (
    <div className="row">
      <div className="modal fade" id="exampleModal" tabIndex={-1} role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
        <FollowingList list={list} />
      </div>
      <div className="modal fade" id="exampleModal1" tabIndex={-1} role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
        <FollowersList followersList={followersList} />
      </div>
      <div className="col-md-4">
        <strong>35 </strong>
      Posts
      </div>
      <div className="col-md-4" data-toggle="modal" data-target="#exampleModal1">
        <strong>
          {followers.follower_count ? followers.follower_count : 0}
          {" "}
        </strong>
        {followers.follower_count === 1 ? "Follower" : "Followers"}
      </div>
      <div className="col-md-4" data-toggle="modal" data-target="#exampleModal">
        <strong>
          {following.following_count ? following.following_count : "0"}
          {" "}
        </strong>
      Following
      </div>
    </div>
  );
};

export default ProfileStats;
