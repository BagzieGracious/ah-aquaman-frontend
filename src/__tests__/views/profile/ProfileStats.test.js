import { shallow } from "enzyme";
import React from "react";
import ProfileStats from "../../../views/profile/ProfileStats";

describe("Home view component", () => {
  it("should render without crashing", () => {
    const wrapper = shallow(<ProfileStats />);
    expect(wrapper).toMatchSnapshot();
  });
});
