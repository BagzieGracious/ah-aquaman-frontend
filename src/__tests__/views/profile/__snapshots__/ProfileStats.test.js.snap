// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view component should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileStats />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-md-4"
        >
          <strong>
            35 
          </strong>
          Posts
        </div>,
        <div
          className="col-md-4"
        >
          <strong>
            767 
          </strong>
          Followers
        </div>,
        <div
          className="col-md-4"
        >
          <strong>
            1,345 
          </strong>
          Following
        </div>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              35 
            </strong>,
            "Posts",
          ],
          "className": "col-md-4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "35 ",
            },
            "ref": null,
            "rendered": "35 ",
            "type": "strong",
          },
          "Posts",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              767 
            </strong>,
            "Followers",
          ],
          "className": "col-md-4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "767 ",
            },
            "ref": null,
            "rendered": "767 ",
            "type": "strong",
          },
          "Followers",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              1,345 
            </strong>,
            "Following",
          ],
          "className": "col-md-4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1,345 ",
            },
            "ref": null,
            "rendered": "1,345 ",
            "type": "strong",
          },
          "Following",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-md-4"
          >
            <strong>
              35 
            </strong>
            Posts
          </div>,
          <div
            className="col-md-4"
          >
            <strong>
              767 
            </strong>
            Followers
          </div>,
          <div
            className="col-md-4"
          >
            <strong>
              1,345 
            </strong>
            Following
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                35 
              </strong>,
              "Posts",
            ],
            "className": "col-md-4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "35 ",
              },
              "ref": null,
              "rendered": "35 ",
              "type": "strong",
            },
            "Posts",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                767 
              </strong>,
              "Followers",
            ],
            "className": "col-md-4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "767 ",
              },
              "ref": null,
              "rendered": "767 ",
              "type": "strong",
            },
            "Followers",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                1,345 
              </strong>,
              "Following",
            ],
            "className": "col-md-4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1,345 ",
              },
              "ref": null,
              "rendered": "1,345 ",
              "type": "strong",
            },
            "Following",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
